src/core/toga/widgets/optioncontainer.py

```diff
@@ -56,10 +113,25 @@ def add(self, label, widget):
         widget.app = self.app
         widget.window = self.window
 
-        self._content.append(widget)
+        self._content.append(label, widget)
         self._impl.add_content(label, widget._impl)
         widget.refresh()
 
+    def remove(self, index):
+
+        disabled_siblings = not any(
+            [
+                opt.enabled for opt in self._content
+                if opt != self._content[index]
+            ]
+        )
```

Comment:
I'm not sure I follow what this logic is for. You can't remove a tab if any of the tabs are disabled? That doesn't seem to make any sense to me? At the very least, an explanatory comment is needed.

Reply:
If we don't have this check and we delete option3 in this scenario:
* option1(disabled)
* option2(disabled)
* option3(enabled)
Remove will work OK but option2 will be focused and disabled at the same time which I think is worse than don't allow remove it. This behaviour at least is happening in macos.

Reply:
Ah - that makes sense, but it's not exactly intuitive. 

I'm not completely convinced "disabling the tab you're on" is as bad as you suggest; but if we *are* going to support this edge case, I'd argue it would make more sense to force focus onto the first non-disabled tab *after* the one you've selected (wrapping around to the first tab if that isn't an option). That would also mean that if there isn't at least one enabled tab to land on, you can't disable a tab, and attempting to do so should raise an error.

As it happens, I was able to get into this state via a different path: in your demo app, if you delete all the tabs but one, and press the "toggle first button", you end up with a "disabled but enabled" tab.

In that state, you can also press the delete tab button, and it does nothing. That's effectively the "nowhere to land" case I described previously.